RedisCaching/CacheContext.cs

```diff
@@ -40,13 +40,13 @@ internal void SetItem<TValue>(CacheItem<TValue> cacheItem)
             client.Database.StringSet(cacheItem.Id, convert.ToRedis(cacheItem), expiry, When.Always, CommandFlags.FireAndForget);
         }
 
-        internal Task SetItemAsync<TValue>(CacheItem<TValue> cacheItem)
+        internal async Task SetItemAsync<TValue>(CacheItem<TValue> cacheItem)
         {
             TimeSpan? expiry = null;
             if (cacheItem.ExpireIn.HasValue)
                 expiry = cacheItem.ExpireIn;
 
-            return client.Database
+            await client.Database
                 .StringSetAsync(cacheItem.Id, convert.ToRedis(cacheItem), expiry, When.Always,
                     CommandFlags.FireAndForget);

```

Comment:
if **there is** a justification to make it async, you have to add 
```suggestion
       CommandFlags.FireAndForget).ConfigureAwait(false);
```
to your await line.